File: apps/mobile/android/src/main/kotlin/com/messenger/store/PrivacyScreenStore.kt
package com.messenger.store

import android.content.Context
import android.content.SharedPreferences
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow

object PrivacyScreenStore {
    private lateinit var prefs: SharedPreferences

    private val _enabled = MutableStateFlow(false)
    val enabled: StateFlow<Boolean> = _enabled

    fun init(context: Context) {
        prefs = context.getSharedPreferences("privacy_screen", Context.MODE_PRIVATE)
        _enabled.value = prefs.getBoolean("enabled", false)
    }

    fun enable() {
        _enabled.value = true
        prefs.edit().putBoolean("enabled", true).apply()
    }

    fun disable() {
        _enabled.value = false
        prefs.edit().putBoolean("enabled", false).apply()
    }
}

File: apps/mobile/ios/Sources/Messenger/store/PrivacyScreenStore.swift
import Foundation

final class PrivacyScreenStore: ObservableObject {
    static let shared = PrivacyScreenStore()
    private static let key = "messenger.privacy_screen.enabled"

    @Published var privacyScreenEnabled: Bool

    private init() {
        privacyScreenEnabled = UserDefaults.standard.bool(forKey: Self.key)
    }

    func enable() {
        privacyScreenEnabled = true
        UserDefaults.standard.set(true, forKey: Self.key)
    }

    func disable() {
        privacyScreenEnabled = false
        UserDefaults.standard.set(false, forKey: Self.key)
    }
}
